spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindRefComparison.java

```diff
@@ -971,6 +1000,39 @@ private void checkRefComparison(Location location, JavaClass jclass, Method meth
         }
     }
 
+    private static boolean comparingEnumsSameType(
+            Type lhsType,
+            Type rhsType) {
+        JavaClass lhsClass = resolveJavaClass(lhsType);
+        JavaClass rhsClass = resolveJavaClass(rhsType);
+        if (lhsClass != null && rhsClass != null) {
+            return lhsClass.isEnum() && rhsClass.isEnum() && lhsClass.equals(rhsClass);
+        }
+        return false;
```

Comment:
As far as I can see this could be simplified like this:
```suggestion
        return lhsClass != null && rhsClass != null && lhsClass.isEnum() && rhsClass.isEnum() && lhsClass.equals(rhsClass);
```

Reply:
it feels to me the version with the `if` is easier to read because it shows that the rest of the check depends on both variables being non-null.